src/Template/Layout/js/app/components/folder-picker/folder-picker.js

```diff
@@ -0,0 +1,72 @@
+import { Treeselect, LOAD_ROOT_OPTIONS, LOAD_CHILDREN_OPTIONS } from '@riophae/vue-treeselect'
+import '@riophae/vue-treeselect/dist/vue-treeselect.css'
+import { t } from 'ttag';
+
+const API_URL = new URL(BEDITA.base).pathname;
+const API_OPTIONS = {
+    credentials: 'same-origin',
+    headers: {
+        'accept': 'application/json',
+    }
+};
+
+export default {
+
+    components: {
+        Treeselect,
+    },
+
+    template: `<div>
+        <Treeselect placeholder="${t`Folder`}" :options="options" :load-options="loadOptions" v-model="value" />
+        <input type="hidden" name="folderSelected" :value="value" />
+    </div>`,
+
+    props: {
+        disabled: {
+            type: Boolean,
+            default: true,
+        },
+    },
+
+    data() {
+        return {
+            options: null,
+            value: null,
+        };
+    },
+
+    methods: {
+
+        async fetchFolders(parent) {
+            const folders = [];
+            let filter = !parent ? 'filter[roots]' : `filter[parent]=${parent.id}`;
+            let page = 1;
+            let i = 0;
+            do {
+                let response = await fetch(`${API_URL}api/folders?${filter}&page=${page}&page_size=100`, API_OPTIONS);
```

Comment:
It seems we are loading `/folders?filter[roots]` always on every index page call
Can we call it only on filter usage? Only when we are using `Copy to/Move to` functionality... 

Calls like `/folders?filter[parent]={id}` are already made upon request....

Reply:
We should instantiate the picker only "on demand". Not easy, now, with the current UI. I'll take a look, however

Reply:
done... when choosing an action (move/copy), the folder roots are fetch.